app/test/testMetadata.js

```diff
@@ -0,0 +1,181 @@
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body.error.should.equal(errorMessages.urlsRequired);
+        done();
+      });
+  });
+
+  it('should raise 400 if missing urls parameter', (done) => {
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body.error.should.equal(errorMessages.urlsRequired);
+        done();
+      });
+  });
+
+  it('should raise 400 if urls is empty', (done) => {
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({urls: []}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body.error.should.equal(errorMessages.urlsRequired);
+        done();
+      });
+  });
+
+  it('should return 200 with metadata for a url', (done) => {
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${exampleMetadata.images[0].url}" />
+            <link rel="icon" href="http://www.example.com/rich-icon.png" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({urls: [goodExampleUrl]}))
+      .end((err, res) => {
+
+        res.should.have.status(200);
+
+        const expectedResponse = {
+          error: '',
+          urls: {}
+        };
+        expectedResponse.urls[goodExampleUrl] = exampleMetadata;
+
+        res.body.should.deep.equal(expectedResponse);
+
+        done();
+      });
+  });
+
+  it('should fallback to favicon.ico if no rich icon found', (done) => {
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${exampleMetadata.images[0].url}" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+
+    const expectedMetadata = JSON.parse(JSON.stringify(exampleMetadata));
+    expectedMetadata.favicon_url = 'http://www.example.com/favicon.ico';
+    delete expectedMetadata.icon_url;
+
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({urls: [goodExampleUrl]}))
+      .end((err, res) => {
+
+        res.should.have.status(200);
+
+        const expectedResponse = {
+          error: '',
+          urls: {}
+        };
+        expectedResponse.urls[goodExampleUrl] = expectedMetadata;
+
+        res.body.should.deep.equal(expectedResponse);
+
+        done();
```

Comment:
Same as above, I think you can do `{ urls: { [goodExampleUrl] : expectedMetadata } }` now in ES6:

```js
      .end((err, res) => {
        const expectedResponse = {
          error: '',
          urls: {
            [goodExampleUrl]: exampleMetadata
          }
        };

        res.should.have.status(200);
        res.body.should.deep.equal(expectedResponse);

        done();
      });
```